### ISSUE TITLE
thin_entrypoint with `--cleanup-config-on-exit=true` is constantly creating and deleting config and kubeconfig

### ISSUE THREAD

**Issue by pmtk** [k8snetworkplumbingwg/multus-cni#1220]
**What happend**:
thin_entrypoint is constantly creating config and kubeconfig. Below is **3s** watch of the operations:
```
$ timeout 3s inotifywait -e modify,create,delete -r /etc/cni/net.d --monitor
Setting up watches.  Beware: since -r was given, this may take a while!
Watches established.
/etc/cni/net.d/multus.d/ CREATE multus.kubeconfig.new
/etc/cni/net.d/multus.d/ MODIFY multus.kubeconfig.new
/etc/cni/net.d/multus.d/ MODIFY multus.kubeconfig.new
/etc/cni/net.d/multus.d/ MODIFY multus.kubeconfig.new
/etc/cni/net.d/multus.d/ MODIFY multus.kubeconfig.new
/etc/cni/net.d/multus.d/ DELETE multus.kubeconfig.new
/etc/cni/net.d/ CREATE 00-multus.conf.new
/etc/cni/net.d/ MODIFY 00-multus.conf.new
/etc/cni/net.d/ MODIFY 00-multus.conf.new
/etc/cni/net.d/ MODIFY 00-multus.conf.new
/etc/cni/net.d/ MODIFY 00-multus.conf.new
/etc/cni/net.d/ MODIFY 00-multus.conf.new
/etc/cni/net.d/multus.d/ CREATE multus.kubeconfig.new
/etc/cni/net.d/multus.d/ MODIFY multus.kubeconfig.new
/etc/cni/net.d/multus.d/ MODIFY multus.kubeconfig.new
/etc/cni/net.d/multus.d/ MODIFY multus.kubeconfig.new
/etc/cni/net.d/multus.d/ DELETE multus.kubeconfig.new
/etc/cni/net.d/ CREATE 00-multus.conf.new
/etc/cni/net.d/ MODIFY 00-multus.conf.new
/etc/cni/net.d/ MODIFY 00-multus.conf.new
/etc/cni/net.d/ MODIFY 00-multus.conf.new
/etc/cni/net.d/ MODIFY 00-multus.conf.new
/etc/cni/net.d/ MODIFY 00-multus.conf.new
/etc/cni/net.d/ MODIFY 00-multus.conf.new
/etc/cni/net.d/ MODIFY 00-multus.conf.new
/etc/cni/net.d/ MODIFY 00-multus.conf.new
/etc/cni/net.d/multus.d/ CREATE multus.kubeconfig.new
/etc/cni/net.d/multus.d/ MODIFY multus.kubeconfig.new
/etc/cni/net.d/multus.d/ MODIFY multus.kubeconfig.new
/etc/cni/net.d/multus.d/ MODIFY multus.kubeconfig.new
/etc/cni/net.d/multus.d/ MODIFY multus.kubeconfig.new
/etc/cni/net.d/multus.d/ DELETE multus.kubeconfig.new
/etc/cni/net.d/ CREATE 00-multus.conf.new
/etc/cni/net.d/ MODIFY 00-multus.conf.new
/etc/cni/net.d/ MODIFY 00-multus.conf.new
/etc/cni/net.d/ MODIFY 00-multus.conf.new
/etc/cni/net.d/ MODIFY 00-multus.conf.new
/etc/cni/net.d/ MODIFY 00-multus.conf.new
/etc/cni/net.d/ MODIFY 00-multus.conf.new
/etc/cni/net.d/ MODIFY 00-multus.conf.new
```

**What you expected to happen**:
thin_entrypoint compares existing 00-multus.conf and multus.kubeconfig in memory or uses different method to decide if files should be updated

**How to reproduce it (as minimally and precisely as possible)**:
Deploy thin multus with `--cleanup-config-on-exit=true` option

**Anything else we need to know?**:

**Environment**:

- Multus version: 
  ```
  ghcr.io/k8snetworkplumbingwg/multus-cni:snapshot
  "Id": "fdd350da2291d12b094c38ff9e7f2dff64d54a0ed42a6db4b9f96edf27157528"
  "Digest": "sha256:329782ee007f10eef918aab8b25e88369adf7e0ecd43231a57012cc81300c6fe"
  ```
- Kubernetes version (use `kubectl version`): v1.29.1
- Primary CNI for Kubernetes cluster: ovn-k
- OS (e.g. from /etc/os-release): rhel9
- File of '/etc/cni/net.d/': not applicable
- File of '/etc/cni/multus/net.d': n/a
- NetworkAttachment info (use `kubectl get net-attach-def -o yaml`): n/a
- Target pod yaml info (with annotation, use `kubectl get pod <podname> -o yaml`)
- Other log outputs (if you use multus logging)

**Comment by s1061123** [comment]
Thank you for the fix, in #1234 (this comment makes association between PR/issue)

**Comment by s1061123** [comment]
Fixed at #1234
